### PR TITLE
Overwriting with mynewt-core upstream + enabling

### PR DESCRIPTION
**Pull request by de-nordic**
_NOTE: Umbrella PR that gathers required changes to other project is here:
https://github.com/zephyrproject-rtos/zephyr/pull/21825_

Overwriting with mynewt-core upstream + changes required for it to run.

Entire code base of tinycbor has been overwritten with mynewt-core  version (https://github.com/apache/mynewt-core) at 13423bde.

Above copy brings work of Christopher Collins, Paul Dietrich, Tim Hutt,
Thiago Macieira, Andrzej Kaczmarek, Marko Kiiskila, Vipul Rahane,
David G. Simmons, Peter Snyder, et al.

### CODE REVIEW

File: src/cborencoder.c [hunk 1]
```diff
@@ -196,34 +194,19 @@
...
-    encoder->writer = w;
-    encoder->remaining = 2;
+    encoder->writer = writer;
+    encoder->added = 0;
     encoder->flags = flags;
 }
 
-
-#ifndef CBOR_NO_FLOATING_POINT
+#if FLOAT_SUPPORT
```
Comment (carlescufi) [inline src/cborencoder.c]: This won't work with Zephyr. We should translate our `CONFIG_FLOAT` to this

File: src/cborparser.c [hunk 2]
```diff
@@ -303,13 +278,13 @@ static CborError preparse_next_value(CborValue *it)
 static CborError advance_internal(CborValue *it)
 {
     uint64_t length;
-    CborError err;
-    err = _cbor_value_extract_number(it->parser, &it->offset,  &length);
-    cbor_assert(err == CborNoError);
+    CborError err = extract_number(it->parser, &it->offset,  &length);
+    err = err;
+    assert(err == CborNoError);
```
Comment (carlescufi) [inline src/cborparser.c]: How does this work with Zephyr?

**Review by carlescufi — CHANGES_REQUESTED** [review]

File: src/cborencoder.c (hunk 1 above)
Comment (de-nordic) [reply src/cborencoder.c]: Yes, I just figured out that I have not tested all possible combinations of Kconfig variables and need to do that. I will update the review shortly.

Comment (nvlsianpu) [reply src/cborencoder.c]: @de-nordic This is not fixed. Am I wrong?

**Review by carlescufi — APPROVED** [review]

**Pull request merged**